src/tokenizer/stop_word_filter/mod.rs

```diff
@@ -31,15 +45,28 @@ impl StopWordFilter {
         }
     }
 
-    fn english() -> StopWordFilter {
-        let words: [&'static str; 33] = [
+    /// Create a `StopWorldFilter` for the named language
+    pub fn english() -> Self {
+        const WORDS: &[&str] = &[
             "a", "an", "and", "are", "as", "at", "be", "but", "by", "for", "if", "in", "into",
             "is", "it", "no", "not", "of", "on", "or", "such", "that", "the", "their", "then",
             "there", "these", "they", "this", "to", "was", "will", "with",
         ];
 
-        StopWordFilter::remove(words.iter().map(|&s| s.to_string()))
+        Self::remove(WORDS.iter().map(|&word| word.to_owned()))
     }
+
+    language!(swedish);
```

Comment:
The macro is no useful here.

```
fn from_words_slice(words: &[&str]) -> Self {
       // ...
}

fn dutch(&self) -> Self {
     Self::from_words_slice(&stopwords::DUTCH)
}

fn italian(&self) -> Self {
     Self::from_words_slice(&stopwords::ITALIAN)
}

...

I think in a separate ticket, we should introduce a Language enum, and rely on that for the stop words filter and the stemmer filter.

Reply:
I think there already is [`tantivy::tokenizer::Language`](https://docs.rs/tantivy/latest/tantivy/tokenizer/enum.Language.html) used by the Stemmer which we could just apply here.

I think the main usability issue here would be that we do not have stop word lists for all of these languages. Should this fail to produce a filter then? Or produce a filter using an empty list?

Reply:
You could return an option yes.

Reply:
Will add that as a separate commit.